### DISCUSSION TITLE
Sunshine should detect for and warn if ViGEmBus is not installed.

### DISCUSSION THREAD

**Discussion by DaddyW123** [LizardByte/community#78] [Sunshine] [answered]
I initially followed a youtube video tutorial by one of the many tech channels that I follow for setting up sunshine.  Apparently he had glazed over the whole "ViGEmBus" requirement for controller use.  I was troubleshooting for several days before I found the document page linked in the Code tab of this github.  And in that page, under troubleshooting, the first item says to make sure that ViGEmBus is installed for controllers to work.
It would have been really nice if either during the Sunshine install, or on the web configuration page, Sunshine detects for this required add-in and warned you that it's missing.

**Comment by ReenigneArcher** [comment]
I don't know what to say. It's well documented that it's required.

Here: https://docs.lizardbyte.dev/projects/sunshine/en/latest/about/usage.html#windows
And Here: https://docs.lizardbyte.dev/projects/sunshine/en/latest/troubleshooting/windows.html

Documentation links are in our readme (as you found), and on our website: https://app.lizardbyte.dev

You could have just sent a quick message to us on Discord (or even here) and avoided wasting several days.

With that said, your feature request is under consideration. Thank you.

**Reply by DaddyW123** [reply]
Hope that didn't come out wrong and offend anyone. I didn't mean my couple of days wasted was your fault.  It was more my own fault for not looking up documentation sooner, and a frustration that the video tutorial (not done by you at all), didn't reference it. So I didn't know I was missing anything until I finally DID look up the documentation.
Yes, once I looked around, it didn't take me long to find it.  But I wanted to make this request because I feel like the feature of detecting it is a solid feature request due to the fact that it is a required add-in.

**Comment by ReenigneArcher** [comment, marked as answer]
For many releases now, the Sunshine installer now takes care of installing vigembus if needed.
